common/test/acceptance/tests/helpers.py

```diff
@@ -354,6 +359,92 @@ def verify_events_of_type(self, username, event_type, expected_events, expected_
+            start_time=start_time, event_filter=event_filter, number_of_matches=number_of_matches)
+
+        if captured_events is not None and hasattr(captured_events, 'append') and callable(captured_events.append):
+            for event in events:
+                captured_events.append(event)
+
+    def wait_for_events(self, start_time=None, event_filter=None, number_of_matches=1):
+        if start_time is None:
+            start_time = self.start_time
+
+        def has_matching_events():
+            matching_events = self.get_matching_events_from_time(start_time=start_time, event_filter=event_filter)
+            return (len(matching_events) >= number_of_matches, matching_events)
+
+        return Promise(
+            has_matching_events,
+            'Waiting for {number_of_matches} events to match the filter.'.format(
+                number_of_matches=number_of_matches
+            )
+        ).fulfill()
+
+    def get_matching_events_from_time(self, start_time=None, event_filter=None):
+        if start_time is None:
+            start_time = self.start_time
+
+        if isinstance(event_filter, dict):
+            event_filter = functools.partial(is_matching_event, event_filter)
+        elif not callable(event_filter):
+            raise ValueError(
+                'event_filter must either be a dict or a callable function with as single "event" parameter that '
+                'returns a boolean value.'
+            )
+
+        matching_events = []
+        cursor = self.event_collection.find(
+            {
+                "time": {
+                    "$gte": start_time
+                }
+            }
+        ).sort("time", ASCENDING)
+        for event in cursor:
+            matches = False
+            try:
+                del event['_id']
+                if event_filter is not None:
+                    matches = event_filter(event)
+            except AssertionError:
+                continue
+            else:
+                if matches is None or matches:
+                    matching_events.append(event)
+        return matching_events
+
+    @contextmanager
+    def assert_matching_events_emitted(self, event_filter, num_events=1):
+        with self.capture_events(event_filter, num_events, []):
+            yield
+
+    def assert_event_sequences_match(self, expected_events, actual_events):
+        for expected_event, actual_event in zip(expected_events, actual_events):
+            assert_event_matches(expected_event, actual_event)
+
+    def assert_no_matching_events_emitted(self, event_filter):
+        matching_events = self.get_matching_events_from_time(event_filter=event_filter)
+        failure_message = [
+            'Matching events were emitted.',
+            'Event Filter:',
+            get_pretty_event_string(event_filter),
+            'Matching Events:',
+        ]
+        for event in matching_events:
+            failure_message.append(get_pretty_event_string(event))
```

Comment:
What is done with failure_message?  Should this be used in the following assertEquals call?